docs/self-managed/zeebe-gateway-deployment/the-zeebe-gateway.md

```diff
@@ -0,0 +1,169 @@
+
+## What is the Zeebe Gateway?
+
+The Zeebe Gateway is a component of the Zeebe cluster; it can be considered the contact point for the Zeebe cluster which allows Zeebe clients to communicate with Zeebe brokers inside a Zeebe cluster. For more information about the Zeebe broker, visit our [additional documentation](../../components/zeebe/technical-concepts/architecture.md#brokers).
+
+To summarize, the Zeebe broker is the main part of the Zeebe cluster, which does all the heavy work like processing, replicating, exporting, and everything based on partitions. The Zeebe Gateway acts as a load balancer and router between Zeebe’s processing partitions.
+
+![Zeebe gateway overview](assets/zeebe-gateway-overview.png)
+
+To interact with the Zeebe cluster, the Zeebe client sends a command as a gRPC message to the Zeebe Gateway (to port `26500` by default). Given the gateway supports gRPC, the user can use several clients in different languages to interact with the Zeebe cluster. For more information, read our [overview](../../apis-clients/overview.md).
+
+When the Zeebe Gateway receives a valid gRPC message, it is translated to an internal binary format and forwarded to one of the partition leaders inside the Zeebe cluster. The command type and values can determine to which partition the command is forwarded.
+
+For example, creating a new process instance is sent in a round-robin fashion to the different partitions. If the command relates to an existing process instance, the command must be sent to the same partition where it was first created (determined by the key).
+
+To determine the current leader for the corresponding partition, the gateway must maintain the topology of the Zeebe cluster. The gateway(s) and broker(s) form a cluster using gossip to distribute information.
+
+## Why do we have it and what problems does it solve?
+
+The Zeebe Gateway protects the brokers from external sources. It allows the creation of a demilitarized zone ([DMZ](https://en.wikipedia.org/wiki/DMZ_(computing))) and the Zeebe Gateway is the only contact point.
+
+The Zeebe Gateway also allows you to easily create clients in your language of choice while keeping the client implementation as thin as possible. There are already several client implementations available, officially-supported, and community-maintained. Check the list [here](../../apis-clients/overview.md).
+
+The gateway can be run and scaled independently of the brokers, which allows separating the concerns of the applications.
+
+## Embedded versus standalone
+
+The Zeebe Gateway can be run in two different ways: embedded and standalone.
+
+### Embedded
+
+Running the gateway in embedded mode means it will run as part of the Zeebe broker. The broker will accept gRPC client messages via the embedded gateway and distribute the translated requests inside the cluster. This means the request accepted by the embedded gateway does not necessarily go to the same broker, where the embedded gateway is running.
```

Comment:
If the request accepted by the embedded gateway does not necessarily go to the same broker, where else would it go?

Reply:
It would go to a different broker, which is the leader of the target partition.

Example: 

You have 3 brokers and each is the leader for one partition:

B0 - Leader of 1
B1 - Leader of 2
B2 - Leader of 3

If we send a command it might be sent to the embedded gateway of B1. If the command is related to a process instance that is executed on partition 1, then the command/message is forwarded to B0.

Does this make sense?